pkg/controller/container-runtime-config/helpers.go

```diff
@@ -172,10 +176,54 @@ func getManagedKeyCtrCfg(pool *mcfgv1.MachineConfigPool) string {
 	return fmt.Sprintf("99-%s-%s-containerruntime", pool.Name, pool.ObjectMeta.UID)
 }
 
+func getStableManagedKeyCtrCfg(pool *mcfgv1.MachineConfigPool, client mcfgclientset.Interface) (string, error) {
+	// if we don't have a client, we're installing brand new, and we don't need to adjust for backward compatibility
+	managedKey := fmt.Sprintf("99-%s-containerruntime", pool.Name)
+	if client == nil {
+		return managedKey, nil
+	}
+	mk := getManagedKeyCtrCfg(pool)
+	oldReg, err := client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), mk, metav1.GetOptions{})
+	if err != nil && !kerr.IsNotFound(err) {
+		return "", fmt.Errorf("could not get MachineConfig %q: %v", mk, err)
+	}
+	// this means no previous CR config were here, so we can start fresh
+	if kerr.IsNotFound(err) {
+		return managedKey, nil
+	}
+	// if we're here, we'll grab the old CR config, dupe it and patch its name
+	oldRegCopy := oldReg.DeepCopy()
+	oldRegCopy.Name = managedKey
+	_, err = client.MachineconfigurationV1().MachineConfigs().Update(context.TODO(), oldRegCopy, metav1.UpdateOptions{})
+	return managedKey, err
+}
+
 func getManagedKeyReg(pool *mcfgv1.MachineConfigPool) string {
 	return fmt.Sprintf("99-%s-%s-registries", pool.Name, pool.ObjectMeta.UID)
 }
 
+func getStableManagedKeyReg(pool *mcfgv1.MachineConfigPool, client mcfgclientset.Interface) (string, error) {
```

Comment:
Optional: Looks like this could be de-duped `getStableManagedKeyCtrCfg` with an internal helper that takes the string suffix.

Reply:
fixed